app/models/concerns/orchestration/dhcp.rb

```diff
@@ -92,8 +92,9 @@ def queue_dhcp_update
 
   # do we need to update our dhcp reservations
   def dhcp_update_required?
-    # IP Address / name changed
-    return true if ((old.ip != ip) or (old.hostname != hostname) or (old.mac != mac) or (old.subnet != subnet))
+    # IP Address / name changed, or 'rebuild' action is triggered and DHCP record on the smart proxy is not present/identical.
+    return true if ((old.ip != ip) or (old.hostname != hostname) or (old.mac != mac) or (old.subnet != subnet) or
+                    (old.build? != build? and build? and !dhcp_record.valid?))
```

Comment:
`old.build? != build? and build?`
I think is the same as
`!old.build? and build?`
which seems more readable to me